web/ui/react-app/src/pages/Services.tsx

```diff
@@ -1,15 +1,116 @@
 import React, { FC } from 'react';
 import { RouteComponentProps } from '@reach/router';
 import PathPrefixProps from '../PathPrefixProps';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
 import { Alert } from 'reactstrap';
+import { useFetch } from '../utils/useFetch';
+import { LabelsTable } from './LabelsTable';
+import { Target, Labels, DroppedTargets } from './targets/target';
 
-const Services: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => (
-  <>
-    <h2>Service Discovery</h2>
-    <Alert color="warning">
-      This page is still under construction. Please try it in the <a href={`${pathPrefix}/service-discovery`}>Classic UI</a>.
-    </Alert>
-  </>
-);
+// TODO: Deduplicate with https://github.com/prometheus/prometheus/blob/213a8fe89a7308e73f22888a963cbf9375217cd6/web/ui/react-app/src/pages/targets/ScrapePoolList.tsx#L11-L14
+interface ServiceMap {
+  activeTargets: Target[];
+  droppedTargets: any[];
+}
+
+export interface TargetLabels {
+  discoveredLabels: Labels;
+  labels: Labels;
+}
+
+const Services: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => {
+  const { response, error } = useFetch<ServiceMap>(`${pathPrefix}/api/v1/targets`);
+
+  const processSummary = (response: ServiceMap) => {
```

Comment:
I would suggest you to split ```proccess``` functions into smaller reducer functions, to pull them out from the component and export them in order to test them. Something like:

```
export const activeTargetsSummary = (acc: any, { scrapePool }: Target) => {............};

export const droppedTargetsSummary = (acc: any, { job }: any) => {..........};

```
Then you can use them like:

```
const { activeTargets, droppedTargets } = response.data;
const activeSummary = activeTargtes.reduce(activeTargetsSummary, {});
const allTargetsSummary = droppedTartgets.reduce(droppedTargetsSummary, activeSummary);
```

Same can be applied for labels even I think both (summary and labels) can be combined into structure like this:
```
{
   active: number;
   total: number;
   labels: Array<{ discoveredLabels: Labels; labels: Labels }>
}
```